src/main/java/org/triplea/client/launch/LaunchScreen.java

```diff
@@ -0,0 +1,654 @@
+    final Dimension size = frame.getSize();
+    frame.getContentPane().removeAll();
+    frame.getContentPane().add(component);
+    frame.setSize(size);
+    frame.pack();
+    frame.setSize(size);
+  }
+
+
+  private JPanel buttonPanel() {
+    return buttonPanel(null);
+  }
+
+  private JPanel buttonPanel(final Runnable backAction) {
+    return buttonPanel(backAction, null);
+  }
+
+  private JPanel buttonPanel(final Runnable backAction, final Runnable playAction) {
+    return JPanelBuilder.builder()
+        .borderEtched()
+        .flowLayout()
+        .add(JButtonBuilder.builder()
+            .biggerFont()
+            .title("Play")
+            .actionListener(playAction == null ? () -> {
+            } : playAction)
+            .enabled(playAction != null)
+            .build())
+        .add(Box.createHorizontalStrut(50))
+        .add(JButtonBuilder.builder()
+            .title("Back")
+            .actionListener(backAction == null ? () -> {
+            } : backAction)
+            .enabled(backAction != null)
+            .build())
+        .add(Box.createHorizontalStrut(50))
+        .add(JButtonBuilder.builder()
+            .title("Quit")
+            .actionListener(() -> {
+              frame.dispose();
+            })
+            .build())
+        .add(Box.createVerticalGlue())
+        .build();
+  }
+
+
+
+  private void showMoreOptions() {
+    replaceContentPaneContents(
+        JPanelBuilder.builder()
+            .addCenter(
+                JPanelBuilder.builder()
+                    .flowLayout()
+                    .add(
+                        JPanelBuilder.builder()
+                            .verticalBoxLayout()
+                            .add(Box.createVerticalStrut(40))
+                            .add(JButtonBuilder.builder()
+                                .title("Settings")
+                                .actionListener(ClientSetting::showSettingsWindow)
+                                .build())
+                            .add(Box.createVerticalStrut(40))
+                            .add(JButtonBuilder.builder()
+                                .title("Map Tools")
+                                .actionListener(() -> ProcessRunnerUtil.runClass(MapCreator.class))
+                                .build())
+                            .add(Box.createVerticalGlue())
+                            .build())
+                    .build())
+            .addSouth(buttonPanel(INSTANCE::showInitialScreen))
+            .build());
+  }
+
+  private void showMultiplayer() {
+    replaceContentPaneContents(
+        JPanelBuilder.builder()
+            .addCenter(
+                JPanelBuilder.builder()
+                    .flowLayout()
+                    .add(
+                        JPanelBuilder.builder()
+                            .verticalBoxLayout()
+                            .add(Box.createVerticalStrut(40))
+                            .add(JButtonBuilder.builder()
```

Comment:
The indentation does get pretty serious at points. Though I think there are some nuances to where it works and it does not. In particular, builder works well when you have configuration code. Some higher level heuristics I think would be better compared to nesting level. The deep nesting can sometimes be a benefit even:
- if something is nested and not a function call, you know you are seeing a unique copy of something, you are free to modify it, no level of indirection, and particularly none when every button and label are off on their own call for a panel that contains many of each
-  avoids a ton of meaningless intermediary names, no more: `buildSouthPanelWithOutPaddingPanel()`
- so long as there is config, it's relatively structured, except within a `builder()` and `build()` block, there is type safety and lines generally cannot be re-arranged safely.

Of interest, FWIW, I've  tried to clean up a number of blocks where the nesting was weird, and found the intermediary variables or single-use methods often did not help. Not to say some places could not be helped, but I would like to be sure we don't just look at indentation level when assessing if something is going to be too complicated.